src/sentry/integrations/jira/notify_action.py

```diff
@@ -4,37 +4,129 @@
 
 logger = logging.getLogger("sentry.rules")
 
 
 class JiraNotifyServiceForm(forms.Form):
-    # TODO 1.0 Add form fields.
+    jira_integration = forms.ChoiceField(choices=(), widget=forms.Select())
 
     def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
         super(JiraNotifyServiceForm, self).__init__(*args, **kwargs)
 
-    def clean(self):
-        return super(JiraNotifyServiceForm, self).clean()
+        if integrations:
+            self.fields["jira_integration"].initial = integrations[0][0]
+
+        self.fields["jira_integration"].choices = integrations
+        self.fields["jira_integration"].widget.choices = self.fields["jira_integration"].choices
 
 
 class JiraCreateTicketAction(TicketEventAction):
     form_cls = JiraNotifyServiceForm
-    label = u"TODO Create a {name} Jira ticket"
+    label = u"""Create a Jira ticket in the {jira_integration} account
+    and {project} project
+    of type {issuetype}
+    with components {components}
+    and due date {duedate}
+    with fixVersions {fixVersions}
+    assigned to {assignee}
+    reported by {reporter}
+    label {labels}
+    priority {priority}"""
     prompt = "Create a Jira ticket"
     provider = "jira"
     integration_key = "jira_integration"
 
     def __init__(self, *args, **kwargs):
         super(JiraCreateTicketAction, self).__init__(*args, **kwargs)
-        # TODO 1.1 Add form_fields
-        self.form_fields = {}
+        integration_choices = [(i.id, i.name) for i in self.get_integrations()]
+
+        if not self.get_integration_id() and integration_choices:
+            self.data[self.integration_key] = integration_choices[0][0]
+
+        self.form_fields = {
+            "jira_integration": {
+                "choices": integration_choices,
+                "default": self.get_integration_id(),
+                "type": "choice",
+                "updatesForm": True,
+            }
+        }
+
+        try:
+            integration = self.get_integration()
+        except Integration.DoesNotExist:
+            return
+
+        installation = integration.get_installation(self.project.organization.id)
+        if installation:
+            try:
+                fields = installation.get_create_issue_config_no_params()
+            except IntegrationError as e:
+                # TODO log when the API call fails.
+                logger.info(e)
+                return
+
+            self.update_form_fields_from_jira_fields(fields)
 
     def render_label(self):
-        return self.label.format(name=self.get_integration_name())
+        """
+        Get the rule as a string. Use human-readable values when available and
+        construct the the label by parts because there are so many optional
+        fields.
+
+        :return: String
+        """
+        labels = ["Create a Jira ticket in the {} account".format(self.get_integration_name())]
+
+        if self.data.get("jira_project"):
```

Comment:
@mgaeta I am concerned with this approach because it is assuming we know all the fields here. There could be other fields that are custom to their integration and we'd be missing that here. Would prefer a more generic approach that would account for custom fields.